src/cata_tiles.h

```diff
@@ -25,9 +25,37 @@ class JsonObject;
 struct visibility_variables;
 
 /** Structures */
+struct sprite_variation {
+    int weight;
+    std::vector<int> sprite_ids;
+
+    sprite_variation()
+    {
+        weight = 1;
```

Comment:
> the default weight of 1 is functional for every case except the first one

Yes, so why are you initializing it to 1 anyway?

> on line 723 I'm creating an "invalid" entry 

I would assume that any default constructed `sprite_variation` is invalid (as it does not contain any sprites, the vector is empty). I see several places that test for `weight == -1`, a default constructed object would not fit there, but should still be considered invalid.

In short: this is inconsistent. If `weight == -1` means the object is invalid and `weight > 0` means it's valid, it should start that way.